Sources/Diagnostics/DiagnosticsEntry.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  DiagnosticsEntry.swift
+//
+//  Created by Cesar de la Vega on 1/4/24.
+
+import Foundation
+
+protocol DiagnosticsEntry: Codable, Equatable {
+    var diagnosticType: String { get }
+    var version: Int { get }
+}
+
+extension DiagnosticsEntry {
+    var version: Int { 1 }
+
+    func toJSONString() -> String? {
+        let encoder = JSONEncoder()
+        encoder.keyEncodingStrategy = .convertToSnakeCase
+        encoder.dateEncodingStrategy = .iso8601
+
+        guard let jsonData = try? encoder.encode(self) else { return nil }
+        return String(data: jsonData, encoding: .utf8)
+    }
```

Comment:
you can simplify this by using `JSONEncoder.default.encode` or even `JSONEncoder.prettyPrinted.encode`. 
Declaration here: https://github.com/RevenueCat/purchases-ios/blob/main/Sources/LocalReceiptParsing/DataConverters/Codable%2BExtensions.swift#L52
Example usage: https://github.com/RevenueCat/purchases-ios/blob/main/Sources/Identity/CustomerInfoManager.swift#L192

Or even better, since `DiagnosticsEntry` is `Codable`, you can leverage the other extension to directly do stuff like calling `prettyPrintedJSON` and `encodedJSON` on the entry itself, i.e.: `let myJSONString = entry.encodedJSON`

Reply:
ooh that's nice!